src/OrdinaryDiffEq.jl

```diff
@@ -265,23 +271,59 @@ SnoopPrecompile.@precompile_all_calls begin
         #AutoVern7(TRBDF2(chunk_size = Val{1}())),
     ]
 
-    prob_list = [ODEProblem{true, SciMLBase.AutoSpecialize}(lorenz, [1.0; 0.0; 0.0],
-                                                            (0.0, 1.0));
-                 ODEProblem{true, SciMLBase.AutoSpecialize}(lorenz, [1.0; 0.0; 0.0],
-                                                            (0.0, 1.0), Float64[])
-                 #ODEProblem{true, SciMLBase.FunctionWrapperSpecialize}(lorenz, [1.0; 0.0; 0.0], (0.0, 1.0));
-                 #ODEProblem{true, SciMLBase.FunctionWrapperSpecialize}(lorenz, [1.0; 0.0; 0.0], (0.0, 1.0), Float64[])
-                 #ODEProblem{true, SciMLBase.NoSpecialize}(lorenz, [1.0; 0.0; 0.0], (0.0, 1.0));
-                 #ODEProblem{true, SciMLBase.NoSpecialize}(lorenz, [1.0; 0.0; 0.0], (0.0, 1.0), Float64[])
-                 #ODEProblem{false,false}(lorenz_oop,[1.0;0.0;0.0],(0.0,1.0))
-                 #ODEProblem{false,false}(lorenz_oop,[1.0;0.0;0.0],(0.0,1.0),Float64[])
-                 ]
+    solver_list = []
+
+    if Preferences.@load_preference("PrecompileNonStiff", true)
```

Comment:
One could enforce booleans here with e.g.
```suggestion
    if Preferences.@load_preference("PrecompileNonStiff", true) === true
```
But I don't know if that's necessary.

Reply:
Doesn't using a value as an expression for an if statement enforce that it should be a boolean?

Reply:
No, Julia will just throw an error. As stated in https://docs.julialang.org/en/v1/manual/control-flow/#man-conditional-evaluation,

> Unlike C, MATLAB, Perl, Python, and Ruby – but like Java, and a few other stricter, typed languages – it is an error if the value of a conditional expression is anything but true or false:

Reply:
yes, it will error saying it needs to be a bool, just like the `=== true`?

Reply:
No, `=== true` will never error as it is guaranteed to always return a boolean. In case `@load_preference` returns `true`, it will be `true`, and otherwise it will be `false`.

Reply:
I see, but it would be good to through an error instead of making it silent when it's incorrectly defined?

Reply:
I guess it is a design choice but I think it would be reasonable to throw an informative error. Probably the only question is if the default error message is clear enough:
```julia
julia> function f(x)
           if x
               return 42
           else
               return -1
           end
       end
f (generic function with 1 method)

julia> f(true)
42

julia> f(1)
ERROR: TypeError: non-boolean (Int64) used in boolean context
Stacktrace:
 [1] f(x::Int64)
   @ Main ./REPL[7]:2
 [2] top-level scope
   @ REPL[9]:1

julia> f("hello")
ERROR: TypeError: non-boolean (String) used in boolean context
Stacktrace:
 [1] f(x::String)
   @ Main ./REPL[7]:2
 [2] top-level scope
   @ REPL[10]:1
```